opt: flake in new benchmark

I saw a failure in the newly-introduced `json-insert` benchmark in `opt/bench/bench_test.go` [here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9083_185.190.8904&logFilter=debug&logView=flowAware).

```
[0:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9083_9083&logFilter=debug&logView=flowAware)      --- FAIL: BenchmarkEndToEnd/json-insert/Simple
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9084_9084&logFilter=debug&logView=flowAware)          bench_test.go:1073: error executing query="INSERT INTO json_table(k, i, j) VALUES ($1, $2, $3)" args=['\x01' '\n' "'{\"a\": \"foo\", \"b\": \"bar\", \"c\": [2, 3, \"baz\", true, false, null]}'"]: pq: error in argument for $3: could not parse JSON: unable to decode JSON: while decoding 65 bytes at offset 0:
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9085_9085&logFilter=debug&logView=flowAware)              ...|'{"a": "foo", "b|...
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9086_9086&logFilter=debug&logView=flowAware)              ...|^...............|...: invalid JSON token
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9087_9087&logFilter=debug&logView=flowAware)      --- FAIL: BenchmarkEndToEnd/json-insert/Prepared
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9088_9088&logFilter=debug&logView=flowAware)          bench_test.go:1087: pq: error in argument for $3: could not parse JSON: unable to decode JSON: while decoding 65 bytes at offset 0:
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9089_9089&logFilter=debug&logView=flowAware)              ...|'{"a": "foo", "b|...
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9090_9090&logFilter=debug&logView=flowAware)              ...|^...............|...: invalid JSON token
[17:50:35 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15349028?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=15349026_9091_9091&logFilter=debug&logView=flowAware)      --- FAIL: BenchmarkEndToEnd/json-insert
```

Jira issue: CRDB-38944

--- a/pkg/sql/opt/bench/bench_test.go
+++ b/pkg/sql/opt/bench/bench_test.go
@@ -455,9 +455,10 @@ var queries = [...]benchQuery{
 		args:  []interface{}{"'abc'"},
 	},
 	{
-		name:  "json-insert",
-		query: "INSERT INTO json_table(k, i, j) VALUES ($1, $2, $3)",
-		args:  []interface{}{1, 10, `'{"a": "foo", "b": "bar", "c": [2, 3, "baz", true, false, null]}'`},
+		name:    "json-insert",
+		query:   `INSERT INTO json_table(k, i, j) VALUES (1, 10, '{"a": "foo", "b": "bar", "c": [2, 3, "baz", true, false, null]}')`,
+		args:    []interface{}{},
+		cleanup: "TRUNCATE TABLE json_table",
 	},
 	{
 		name: "batch-insert-one",

